drivers/mtd/spi/winbond-otp.c

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2015 Imagination Technologies
+ * Author: Avinash Tahakik <avinash.tahakik@imgtec.com>
+ *
+ * SPDX-License-Identifier:	GPL-2.0+
+ */
+
+#include <common.h>
+#include <spi.h>
+#include <spi_flash.h>
+#include <winbond-otp.h>
+
+#include "sf_internal.h"
+
+#define SECURITY_REG_START_ADDR		0x1000 /*first security register addr*/
+#define SECURITY_REG_ADDR_OFFSET	0x1000 /*diff between consecutive reg*/
+#define SECURITY_REG_NUM		3 /* number of security registers */
+#define SECURITY_REG_SIZE		256 /* bytes per security register */
+#define SECURITY_REG_TOTAL_SIZE		(SECURITY_REG_NUM * SECURITY_REG_SIZE)
+#define SPI_NOR_UNIQUE_ID_LEN		8 /*number of bytes of unique ID */
+
+/* SPI FLASH opcodes */
+#define SPINOR_OP_PR_SECURITY_REG	0x42 /* Program security register */
+#define SPINOR_OP_ER_SECURITY_REG	0x44 /* Erase security register */
+#define SPINOR_OP_RD_SECURITY_REG	0x48 /* Read security register */
+
+int read_otp_data(loff_t from, size_t len, char *data) {
+	struct spi_flash *flash = spi_flash_probe(1, 0, CONFIG_SF_DEFAULT_SPEED, CONFIG_SF_DEFAULT_MODE);
```

Comment:
This is the old API, but we use the new driver model for accessing the flash and we define it in the device tree so would spi_flash_probe_fdt not be better?